scripts/dependencies/tar-archives.ts

```diff
@@ -0,0 +1,182 @@
+      await addEntry(`${ layerHash }/json`, Buffer.from(JSON.stringify({
+        id:     layerHash,
+        config: {
+          ExposedPorts: { '80/tcp': {} },
+          WorkingDir:   '/',
+          Entrypoint:   [`/${ path.basename(executablePath) }`],
+        },
+      })));
+      await addEntry(`${ layerHash }.json`, Buffer.from(JSON.stringify({
+        architecture: context.isM1 ? 'arm64' : 'amd64',
+        config:       {
+          ExposedPorts: { '80/tcp': {} },
+          Entrypoint:   [`/${ path.basename(executablePath) }`],
+          WorkingDir:   '/',
+        },
+        history: [],
+        os:      'linux',
+        rootfs:  {
+          type:     'layers',
+          diff_ids: [`sha256:${ layerHash }`],
+        },
+      })));
+      await addEntry('manifest.json', Buffer.from(JSON.stringify([
+        {
+          Config:   `${ layerHash }.json`,
+          RepoTags: ['ghcr.io/rancher-sandbox/rancher-desktop/rdx-proxy:latest'],
+          Layers:   [`${ layerHash }/layer.tar`],
+        },
+      ])));
+      image.finalize();
+      await imageWritten;
+      console.log('Built RDX port proxy image');
+    } finally {
+      await fs.promises.rm(workDir, { recursive: true });
+    }
+  }
+
+  getAvailableVersions(includePrerelease?: boolean | undefined): Promise<string[] | AlpineLimaISOVersion[]> {
+    throw new Error('extension-proxy does not have versions.');
+  }
+
+  rcompareVersions(version1: string | AlpineLimaISOVersion, version2: string | AlpineLimaISOVersion): 0 | 1 | -1 {
+    throw new Error('extension-proxy does not have versions.');
+  }
+}
+
+export class WSLDistroImage implements Dependency {
+  name = 'WSLDistroImage';
+  dependencies(context: DownloadContext): string[] {
+    return ['WSLDistro:win32', 'guestagent:linux'];
+  }
+
+  async download(context: DownloadContext): Promise<void> {
+    const tarName = `distro-${ context.versions.WSLDistro }.tar`;
+    const pristinePath = path.join(context.resourcesDir, context.platform, 'staging', tarName);
+    const pristineFile = fs.createReadStream(pristinePath);
+    const extractor = tar.extract();
+    const destPath = path.join(context.resourcesDir, context.platform, tarName);
+    const destFile = fs.createWriteStream(destPath);
+    const packer = tar.pack();
+
+    console.log('Building WSLDistro image...');
+
+    // Copy the pristine tar file to the destination.
+    packer.pipe(destFile);
+    extractor.on('entry', (header, stream, callback) => {
+      stream.pipe(packer.entry(header, callback));
+    });
+    await stream.promises.finished(pristineFile.pipe(extractor));
+
+    async function addFile(fromPath: string, name: string, options: Omit<tar.Headers, 'name' | 'size'> = {}) {
+      const { size } = await fs.promises.stat(fromPath);
+      const inputFile = fs.createReadStream(fromPath);
+
+      console.log(`WSL Distro: Adding ${ fromPath } to ${ name }...`);
```

Comment:
nit; I noticed on line [172](https://github.com/rancher-sandbox/rancher-desktop/pull/7145/files#diff-cfbdcd170f33b743f046eccb7a5fbd2df72875395efac2e9c033748b87896fe2R172) it is spelled `WSLDistro` and here is `WSL Distro`, we should probably just use one of them to unify the logs.